js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+                const isBuyer = this.safeValue (trade, 'isBuyer');
+                const isMaker = this.safeValue (trade, 'isMaker');
+                const buyerMaker = this.safeString2 (trade, 'isBuyerMaker', 'm');
+                if (isMaker !== undefined) {
+                    takerOrMaker = isMaker ? 'maker' : 'taker';
+                }
+                if (isBuyer !== undefined) {
+                    side = isBuyer ? 'buy' : 'sell';
+                }
+                if (buyerMaker !== undefined) {
+                    side = buyerMaker ? 'sell' : 'buy';
+                    takerOrMaker = 'taker';
+                }
+                const feeAsset = this.safeString (trade, 'commissionAsset');
+                if (feeAsset !== undefined) {
+                    fee = {
+                        'cost': this.safeNumber (trade, 'commission'),
+                        'currency': this.safeCurrencyCode (feeAsset),
+                    };
+                }
+            }
+        }
+        if (id === undefined) {
+            id = this.syntheticTradeId (market, timestamp, side, amountString, priceString, type, takerOrMaker);
+        }
+        return this.safeTrade ({
+            'id': id,
+            'order': orderId,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'type': type,
+            'side': side,
+            'takerOrMaker': takerOrMaker,
+            'price': priceString,
+            'amount': amountString,
+            'cost': costString,
+            'fee': fee,
+            'info': trade,
+        }, market);
+    }
+
+    syntheticTradeId (market = undefined, timestamp = undefined, side = undefined, amount = undefined, price = undefined, orderType = undefined, takerOrMaker = undefined) {
+        // TODO: can be unified method? this approach is being used by multiple exchanges (mexc, woo-coinsbit, dydx, ...)
+        let id = '';
+        if (timestamp !== undefined) {
+            id = this.numberToString (timestamp) + '-' + this.safeString (market, 'id', '_');
+            if (side !== undefined) {
+                id += '-' + side;
+            }
+            if (amount !== undefined) {
+                id += '-' + this.numberToString (amount);
+            }
+            if (price !== undefined) {
+                id += '-' + this.numberToString (price);
+            }
+            if (takerOrMaker !== undefined) {
+                id += '-' + takerOrMaker;
+            }
+            if (orderType !== undefined) {
+                id += '-' + orderType;
+            }
+        }
+        return id;
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const options = this.safeValue (this.options, 'timeframes', {});
+        const timeframes = this.safeValue (options, market['type'], {});
```

Comment:
This timeframe logic is currently broken because the  "timeframes" object is not yet indexed by market type

https://github.com/ccxt/ccxt/pull/12376/files#diff-321f459daa26ff101cea1bedf2b82d023dcbb2422700f94579df4e193e1323caR273